src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -294,6 +294,33 @@ public IEnumerable<NuGetProject> GetNuGetProjects()
             return projects;
         }
 
+        public int GetExpectedProjectsNomination()
+        {
+#if VS14
+            // for VS14, always return 0 since nominations don't apply there.
+            return 0;
+#else
+            var cpsProjects = GetNuGetProjects().OfType<CpsPackageReferenceProject>().ToList();
+
+            // count for nominated projects so far
+            int nominatedProjects = 0;
+
+            foreach (var cpsProject in cpsProjects)
+            {
+                // check if this .Net core project is nominated or not, and accordingly increase counter.
+                DependencyGraphSpec projectRestoreInfo;
+                if (_projectSystemCache.TryGetProjectRestoreInfo(cpsProject.MSBuildProjectPath, out projectRestoreInfo) &&
```

Comment:
`TryGetProjectRestoreInfo` is only being used in CPSProjectReferenceProject and it always passes projectFullPath which is why I passed the same thing here to have the consistency.